src/subhalo_tracking.cpp

```diff
@@ -864,19 +864,51 @@ class SubhaloMasker_t
     if (subhalo.Nbound <= 1)
       return; // skip orphans
 
+    // Want to identify first tracer so we can add it back if all tracers are masked out
+    // Need to recalculate tracer index due to gas particles disappearing
+    // For DMO runs IsTracer will return 1 for all particles anyway
+    // TODO: Is this needed or will Victor set tracerIndex?
+    auto tracer = subhalo.Particles[0];
+    for (auto it = subhalo.Particles.begin(); it != subhalo.Particles.end(); ++it)
```

Comment:
I think this will iterate over any unbound particles too. If we're intending that all subhalos have >0 bound tracers then it should only matter if something has gone wrong. But maybe the code below should throw an exception if there's no tracer in just the bound part of the subhalo?

Reply:
Do you mean convert the subhalo to an orphan at this point if no bound tracers are found? From Victor's examples it is possible to for all the bound tracers to be masked out, so I don't want to abort the run in that instance.

Reply:
Sorry, I hadn't understood this properly. I don't think Nbound is even valid at this point so it does make sense to iterate over all of the particles.